src/modules/cookieHunter/cookieHunter.helpers.ts

```diff
@@ -0,0 +1,187 @@
+    await reaction.message.reply({
+      content: `Il est lent ce lait... 🥛`,
+      options: { ephemeral: true },
+    });
+  } else {
+    await cache.set('milkJokerUserId', user.id);
+    await reaction.message.reply({
+      content: `Premier arrivé, premier servit. Cul sec 🥛 !`,
+      options: { ephemeral: true },
+    });
+  }
+};
+
+const applyMilkJoker = async () => {
+  const milkJokerUserId = await cache.get('milkJokerUserId');
+  if (!milkJokerUserId) return;
+
+  const cookieHunterDailyCount = await cache.get('cookieHunterDailyCount', {});
+  const userDailyCount = cookieHunterDailyCount[milkJokerUserId] || 0;
+  const newDailyCount = { ...cookieHunterDailyCount, [milkJokerUserId]: userDailyCount * 2 };
+  await cache.set('cookieHunterDailyCount', newDailyCount);
+};
+
+const logDailyCount = async (client: Client<true>) => {
+  const dailyLogChannels = await cache.get('cookieHunterDailyLogChannels', []);
+  if (!dailyLogChannels.length) return;
+
+  const currentHuntMessageId = await cache.get('currentHuntMessageId');
+  if (!currentHuntMessageId)
+    throw new Error('Lost the hunt message id before logging the daily count');
+
+  const cookieHunterDailyCount = await cache.get('cookieHunterDailyCount', {});
+  const hunterCount = Object.keys(cookieHunterDailyCount).length - 1; // grandma is not a hunter
+
+  const milkJokerUserId = await cache.get('milkJokerUserId');
+
+  const resume = `**🍪 Résumé de la chasse aux cookies du jour**\n`;
+  const where = `Mamie a servi des cookies dans <#${currentHuntMessageId}>\n`;
+  const baseMessage = `${resume}${where}`;
+
+  const message =
+    hunterCount > 0
+      ? getHuntersFoundGrandmaMessage(baseMessage, cookieHunterDailyCount, milkJokerUserId)
+      : `${baseMessage}**🍪 Personne n'a trouvé Mamie !**\nFaut dire qu'elle se cache bien (et que vous êtes nazes) !`;
+
+  for (const channelId of dailyLogChannels) {
+    const channel = await client.channels.fetch(channelId);
+    if (!channel || !channel.isTextBased()) return;
+    await channel.send(message);
+  }
+};
+
+const getHuntersFoundGrandmaMessage = (
+  baseMessage: string,
+  cookieHunterDailyCount: Record<string, number>,
+  milkJokerUserId: string | undefined,
+) => {
+  const cookieEatenCount = Object.values(cookieHunterDailyCount).reduce(
+    (acc, count) => acc + count,
+    0,
+  );
+  const dailyRank = Object.entries(cookieHunterDailyCount).sort((a, b) => b[1] - a[1]);
+
+  const totalEaten = `Nombre de cookies total mangés : ${cookieEatenCount}\n`;
+  const ranking = `**Classement des chasseurs de cookies du jour**\n`;
+  const usersRanking = dailyRank.map(([userId, count]) => `<@${userId}>: ${count}\n`).join('\n');
+  const milkJoker = milkJokerUserId
+    ? `<@${milkJokerUserId}> a accompagné ses cookies d'un grand verre de lait 🥛\n`
+    : '';
+  const lastWords = `Sacré bande de gourmands !`;
+
+  return `${baseMessage}${totalEaten}${ranking}${usersRanking}${milkJoker}${lastWords}`;
```

Comment:
```suggestion
  return [
    baseMessage,
    `Nombre de cookies total mangés : ${cookieEatenCount}`,
    '**Classement des chasseurs de cookies du jour**',
    ...dailyRank.map(([userId, count]) => `<@${userId}>: ${count}`),
    milkJokerUserId ? `<@${milkJokerUserId}> a accompagné ses cookies d'un grand verre de lait 🥛` : '',
    'Sacré bande de gourmands !',
  ].join('\n');`;
```

probably more readable?

Reply:
yes it's better, thanks